File: OP_laba5_part2_sharp/OP_laba5_part2_sharp/UpperCaseLetters.cs
using System;
using System.Linq;
    
namespace OP_laba5_part2_sharp
{
    public class UpperCaseLetters : ParentString
    {
        public UpperCaseLetters(params char[] str) : base()
        {
            if (!str.All(Char.IsUpper))
            {
                throw new Exception("String should only contain capital letters!");
            }
        }
        public override void Sort()
        {
            Array.Sort(StrValue);
        }
    }
}

File: OP_laba5_part2_c++/src/UpperCaseLetters.cpp
#include "UpperCaseLetters.h"


UpperCaseLetters::UpperCaseLetters(char *value) : ParentString(value) {
    int length = sizeof(value);
    int upper = 0;
    for (int i = 0; i < length; ++i) {
        if(std::isupper(value[i])) {
            upper++;
        }
    }
    if (!(length == upper))
    {
        throw std::runtime_error("String should only contain uppercase letters!");
    }
}

void UpperCaseLetters::Sort()
{
    std::sort(value, value+sizeof(value));

}
